### ISSUE TITLE
Add `isCompleted` to `Upload` info class

### ISSUE THREAD

**Issue by Murderlon** [tus/tus-node-server#459]
This could be useful for lifecycle rules in S3 to clean up incomplete parts and in general.

**Comment by Acconut** [comment]
I am not sure if an entry inside the JSON content of the info object would help here (that's how I understand your proposal). AFAIK, S3 lifecycle rules are not able to access the content of files stored on S3 and can thus not filter out files that are missing the `isCompleted` property.

S3 tags might help here: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html Each completed upload can get a `isCompleted` tag and then the lifecycle rules can filter out files that don't have this tag but are too old.

**Comment by fenos** [comment]
Yes, this should be part of a tag in order for lifecycle rules to pick the change

**Comment by fenos** [comment]
@mitjap incompleteParts are uploaded as real objects in S3 not as parts, hence this was required. However, i think that using the expiration date is cleaner as implemented here #513

**Comment by Murderlon** [comment]
Closed in https://github.com/tus/tus-node-server/pull/513
